src/lazy/text/value.rs

```diff
@@ -46,7 +48,10 @@ impl<'data> LazyRawValue<'data, TextEncoding> for LazyRawTextValue<'data> {
     }
 
     fn read(&self) -> IonResult<RawValueRef<'data, TextEncoding>> {
-        let matched_input = self.input.slice(0, self.encoded_value.data_length());
+        let matched_input = self.input.slice(
+            self.encoded_value.data_offset() - self.input.offset(),
+            self.encoded_value.data_length(),
+        );
```

Comment:
🗺️ Prior to this PR, the matched input slice always started with the value. Now that we have struct fields, it sometimes begins at the field name.